core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -237,6 +237,7 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
   def jars: Seq[String] = _jars
   def files: Seq[String] = _files
   def master: String = _conf.get("spark.master")
+  def deployMode: String = _conf.getOption("spark.submit.deployMode").getOrElse("client")
```

Comment:
I'd just like to point out that this is a public API that we won't be able to remove easily in the future. This is probably OK since I don't see the concept of deploy mode going away ever, but this is worth a mention.